code/game/machinery/rechargestation.dm

```diff
@@ -102,9 +105,21 @@
 			R.adjustFireLoss(-wire_rate)
 	else if(ishuman(occupant))
 		var/mob/living/carbon/human/H = occupant
+		var/obj/item/organ/coolantpump/CP = H.internal_organs_by_name["coolant pump"]
 		if(!isnull(H.internal_organs_by_name["cell"]) && H.nutrition < H.max_nutrition)
 			H.adjustNutritionLoss(-10)
 			cell.use(7000/H.max_nutrition*10)
+		if(!CP)
+			return
+		else
+			if(beaker)
+				for (var/datum/reagent/R in beaker.reagents.reagent_list)

```

Comment:
I assume reagent_list would only contain reagent types, so you can make this loop `for (var/a in beaker.reagents.reagent_list)` and then just have line `var/datum/reagent/R = a` inside of `if (istype(R, /datum/reagent/coolant))` check, this will save in-built istype check that for each loop has in byond. Also you can just put it inside your your